diff --git a/src/Bootstrap.php b/src/Bootstrap.php
index 235a717..0cfc92f 100644
--- a/src/Bootstrap.php
+++ b/src/Bootstrap.php
@@ -54,9 +54,9 @@ class Bootstrap {
 			$configuration = [];
 			$configuration[ 'IP' ] = $GLOBALS[ 'IP' ];
 			$configuration[ 'remoteBasePath' ] =
-				$GLOBALS[ 'wgExtensionAssetsPath' ] . '/Bootstrap/resources/bootstrap';
+				$GLOBALS[ 'wgScriptPath' ] . '/canasta-extensions/Bootstrap/resources/bootstrap';
 			$configuration[ 'localBasePath' ] =
-				$GLOBALS[ 'wgExtensionDirectory' ] . '/Bootstrap/resources/bootstrap';
+				$GLOBALS[ 'IP' ] . '/canasta-extensions/Bootstrap/resources/bootstrap';
 
 			$setupAfterCache = new SetupAfterCache( $configuration );
 			$setupAfterCache->process();
