diff --git a/src/Hooks/SetupAfterCache.php b/src/Hooks/SetupAfterCache.php
index 169bfa5..3d2c002 100644
--- a/src/Hooks/SetupAfterCache.php
+++ b/src/Hooks/SetupAfterCache.php
@@ -92,9 +92,9 @@ class SetupAfterCache {
 	 */
 	protected function setInstallPaths() {
 		$this->configuration[ 'chameleonLocalPath' ] =
-			$this->configuration['wgStyleDirectory'] . '/chameleon';
+			$this->configuration['IP'] . '/canasta-skins/chameleon';
 		$this->configuration[ 'chameleonRemotePath' ] =
-			$this->configuration['wgStylePath'] . '/chameleon';
+			$this->configuration['wgScriptPath'] . '/canasta-skins/chameleon';
 	}
 
 	protected function addLateSettings() {
