diff --git a/includes/CommentStreamsHooks.php b/includes/CommentStreamsHooks.php
index f89e429..ccf2aad 100644
--- a/includes/CommentStreamsHooks.php
+++ b/includes/CommentStreamsHooks.php
@@ -355,7 +355,7 @@ class CommentStreamsHooks {
 	 */
 	public static function showSearchHitTitle(
 		Title &$title,
-		?string &$text,
+		&$text,
 		SearchResult $result,
 		array $terms,
 		SpecialSearch $page
