diff --git a/libs/PF_submit.js b/libs/PF_submit.js
index 600a96e1..083ae7f8 100644
--- a/libs/PF_submit.js
+++ b/libs/PF_submit.js
@@ -215,7 +215,7 @@
 				// Special submit form & other actions handling when VEForAll editor is present
 				if ( $('.visualeditor').length > 0 ) {
 					// Interrupt "Save page" and "Show changes" actions
-					var $formButtons = $( '#wpSave, #wpDiff' );
+					var $formButtons = $( 'button[name="wpSave"], button[name="wpDiff"]' );
 					var canSubmit = false;
 
 					if ( $formButtons.length > 0 ) {
